harness/determined/pytorch/_pytorch_context.py

```diff
@@ -62,6 +100,206 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._last_backward_batch_idx = None  # type: Optional[int]
         self._current_batch_idx = None  # type: Optional[int]
 
+        self._wrapped_reducers = []  # type: List[_WrappedReducer]
+        self._allgather_fn = default_allgather_fn
+
+    def _set_allgather_fn(self, fn: Callable) -> None:
+        self._allgather_fn = fn
+
+    def reset_reducers(self) -> None:
+        for wrapped in self._wrapped_reducers:
+            wrapped.reducer.reset()
+
+    def wrap_reducer(
```

Comment:
Actually, if you provide a function reducer you definitely don't get the same thing back, so I do think that `wrap_reducer` is the right name.

> Sounds weird if we need to wrap our own-defined reducer

I'm not sure what you mean by this?  We only wrap reducers provided by the user.

Reply:
Do you mean: "wrap_reducer" seems weird, since "wrap_model" is how we take a pytorch-native model, but since there is no pytorch-native reducer we shouldn't have to wrap it?

Reply:
I think we are wrapping it with additional functionality, and in some cases wrapping a function in an entire class.  So I'm ok with `wrap_reducer`.